service/cloudconfig/v_0_1_0_master_template.go

```diff
@@ -137,6 +140,12 @@ func (e *v_0_1_0MasterExtension) Files() ([]k8scloudconfig.FileAsset, error) {
 			Owner:        FileOwner,
 			Permissions:  FilePermission,
 		},
+		{
+			AssetContent: e.keys.APIServerEncryptionKey,
+			Path:         "/etc/kubernetes/encryption/k8s-encryption-config.yaml.enc",
+			Owner:        FileOwner,
+			Permissions:  0644,
```

Comment:
please use more strict permissions e.g. 0600

Reply:
Ok. I've mapped to the config file permission

```
ls -lrth /etc/kubernetes/config/
total 16K
-rw-r--r--. 1 root root 392 Oct 12 12:05 proxy-kubeconfig.yml
-rw-r--r--. 1 root root 396 Oct 12 12:05 kubelet-kubeconfig.yml
```

Reply:
kubeconfig itself is ok, as it does not contain (in our) any private keys. But i see that even private keys have insecure permissions on host cluster.

```
r7vme@ip-10-0-5-126 ~ $ ls -la /etc/kubernetes/ssl
total 64
drwxr-xr-x. 4 root root 4096 Sep 13 19:58 .
drwxr-xr-x. 6 root root 4096 Sep 13 19:59 ..
-rw-r--r--. 1 root root 1219 Sep 13 19:58 apiserver-ca.pem
-rw-r--r--. 1 root root 1459 Sep 13 19:58 apiserver-crt.pem
-rw-r--r--. 1 root root 1674 Sep 13 19:58 apiserver-key.pem
drwxr-xr-x. 2 root root 4096 Sep 13 19:56 calico
drwxr-xr-x. 2 root root 4096 Sep 13 19:56 etcd
-rw-r--r--. 1 root root 1651 Sep 13 19:58 service-account-key.pem
```

This means that anyone who has access to node will be able to read private key.

btw, i've checked guest clusters and they have correct permissions
```
r7vme@ip-10-0-5-126 ~ $ ls -la /etc/kubernetes/ssl/etcd/
total 64
drwxr-xr-x. 2 root root 4096 Sep 13 19:56 .
drwxr-xr-x. 4 root root 4096 Sep 13 19:58 ..
-rw-r--r--. 1 root root 1219 Sep 13 19:56 client-ca.pem
-rw-r--r--. 1 root root 1333 Sep 13 19:56 client-crt.pem
-rw-r--r--. 1 root root 1678 Sep 13 19:56 client-key.pem
-rw-r--r--. 1 root root 1219 Sep 13 19:56 server-ca.pem
-rw-r--r--. 1 root root 1333 Sep 13 19:56 server-crt.pem
-rw-r--r--. 1 root root 1678 Sep 13 19:56 server-key.pem
```

So far you need to use strict permissions. And we also need to fix them for [host clusters](https://github.com/giantswarm/giantswarm/issues/1680#issuecomment-336806139).

Reply:
After discussion here https://github.com/giantswarm/giantswarm/issues/1680

makes sense to just use default strict `Permissions:  FilePermission,`

Reply:
yep for sure lets use 600 for the encrypt key